vcr/pe/presentation_submission.go

```diff
@@ -115,43 +115,69 @@ func (b *PresentationSubmissionBuilder) Build(format string) (PresentationSubmis
 
 	// next we need to map the selected VCs to the correct wallet
 	// loop over all selected VCs and find the wallet that contains the VC
-	for j := range selectedVCs {
-		for i, walletVCs := range b.wallets {
-			var index int
-			for _, walletVC := range walletVCs {
-				// do a JSON equality check
-				if vcEqual(selectedVCs[j], walletVC) {
-					signInstructions[i].Holder = b.holders[i]
-					signInstructions[i].VerifiableCredentials = append(signInstructions[i].VerifiableCredentials, selectedVCs[j])
-					// remap the path to the correct wallet index
-					inputDescriptorMappingObjectForVC := inputDescriptorMappingObjects[j]
-					inputDescriptorMappingObjectForVC.Path = fmt.Sprintf("$.verifiableCredential[%d]", index)
-					signInstructions[i].inputDescriptorMappingObjects = append(signInstructions[i].inputDescriptorMappingObjects, inputDescriptorMappingObjectForVC)
-					index++
+	// there will be at most len(b.wallets) sign instructions
+	var signInstructions []SignInstruction
+	for _, holderID := range b.holders {
+		var signInstruction *SignInstruction
+		// find VCs that need to be in this sign instruction/verifiable presentation
+		for i, selectedVC := range selectedVCs {
+			if b.isHolder(holderID, selectedVC) {
```

Comment:
This changes logic. Before any selected VC from a wallet would be added regardless of the holder. Currently we said that all VCs need to have the same subject/holder, but if that changes this selection method won't work. But if `isHolder` is changed back to just checking the Raw()/Equal then it would be possible again. Summary: the holderID == subject check does not prevent false positives but does block future combination of VCs.

Reply:
reverted and fixed bugs in original code